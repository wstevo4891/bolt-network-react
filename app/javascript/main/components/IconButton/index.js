// Icon Button Component

import React from 'react'
import PropTypes from 'prop-types'

import makeToggleable from '@helpers/makeToggleable'

import Button from '../Button'
import Icon from '../Icon'

function buildIconProps(icon, iconProps) {
  if (icon) {
    return Object.assign({}, iconProps, { icon })
  }

  return iconProps
}

function buildButtonProps(buttonProps, handleClick) {
  if (buttonProps.handleClick) return buttonProps

  return Object.assign({}, buttonProps, { handleClick })
}

const IconButton = (props) => {
  const propsForButton = buildButtonProps(props.buttonProps, props.handleClick)
  const propsForIcon = buildIconProps(props.icon, props.iconProps)

  return (
    <Button {...propsForButton}>
      {props.textPlacement === 'left' && props.text}
      <Icon {...propsForIcon} />
      {props.textPlacement === 'right' && props.text}
    </Button>
  ) 
}

IconButton.propTypes = {
  buttonProps: PropTypes.shape(Button.propTypes),
  handleClick: PropTypes.func,
  icon: PropTypes.string,
  iconProps: PropTypes.shape(Icon.propTypes),
  text: PropTypes.string,
  textPlacement: PropTypes.oneOf(['left', 'right']),
}

IconButton.defaultProps = {
  buttonProps: {},
  handleClick: () => void {},
  icon: null,
  iconProps: null,
  text: null,
  textPlacement: 'right',
}

const ToggleIconButton = makeToggleable(IconButton)

export {
  IconButton,
  ToggleIconButton
}
