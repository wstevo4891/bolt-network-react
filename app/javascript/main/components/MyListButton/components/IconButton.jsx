// MyList IconButton

import React from 'react'

const IconButton = (props) => (
  <li className="poster-btn poster-btn-my-list">
    <button onClick={props.handleClick}>
      <i className={props.iconClass}></i>
    </button>
  </li>
)

export default IconButton
