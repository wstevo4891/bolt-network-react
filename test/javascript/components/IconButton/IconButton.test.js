import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import { IconButton } from '@components/IconButton'

describe('IconButton', () => {
  const myListClick = jest.fn()

  it('renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(
      <IconButton
        buttonClass="btn-clear"
        handleClick={myListClick}
        icon="fa-plus"
        text="MY LIST"
      />,
      div
    )

    ReactDOM.unmountComponentAtNode(div)
  })

  it('should render correctly in "debug" mode', () => {
    const component = shallow(
      <IconButton
        buttonClass="btn-clear"
        handleClick={myListClick}
        icon="fa-plus"
        text="MY LIST"
      />
    )
  
    expect(component).toMatchSnapshot()
  })

  it('should respond to click event', () => {
    const component = shallow(
      <IconButton
        buttonClass="btn-clear"
        handleClick={myListClick}
        icon="fa-plus"
        text="MY LIST"
      />
    )

    component.find('button').simulate('click')

    expect(myListClick.mock.calls.length).toEqual(1)
  })
})
