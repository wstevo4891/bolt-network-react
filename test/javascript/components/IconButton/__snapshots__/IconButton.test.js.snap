// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render correctly in "debug" mode 1`] = `
<button
  className="btn-clear"
  onClick={[MockFunction]}
>
  <i
    className="fa fa-plus"
  />
  MY LIST
</button>
`;
